### PR TITLE
Add build_all tests for flash, bbram and eeprom

### PR DESCRIPTION
**Pull request by soburi**
The flash and bbram tests target the following devices at this time.

- microchip,mcp7940n
- atmel,at45
- jedec,spi-nor

Add the following device to the EEPROM test.

- ti,tmp116-eeprom

### CODE REVIEW

File: tests/drivers/build_all/bbram/src/main.c [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright (c) 2024 TOKITA Hiroshi
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+int main(void)
+{
+	return 0;
+}
```
Comment (henrikbrixandersen) [inline tests/drivers/build_all/bbram/src/main.c:9]: You can leave this out.

File: tests/drivers/build_all/bbram/testcase.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2024 TOKITA hiroshi
+# SPDX-License-Identifier: Apache-2.0
+
+tests:
```
Comment (henrikbrixandersen) [inline tests/drivers/build_all/bbram/testcase.yaml:4]: Please add a `common:` section for the common fields.

File: tests/drivers/build_all/flash/src/main.c [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright (c) 2024 TOKITA Hiroshi
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+int main(void)
+{
+	return 0;
+}
```
Comment (henrikbrixandersen) [inline tests/drivers/build_all/flash/src/main.c:9]: Same here.

File: tests/drivers/build_all/flash/testcase.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2024 TOKITA hiroshi
+# SPDX-License-Identifier: Apache-2.0
+
+tests:
```
Comment (henrikbrixandersen) [inline tests/drivers/build_all/flash/testcase.yaml:4]: and same here.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: tests/drivers/build_all/bbram/testcase.yaml (hunk 2 above)
Comment (soburi) [reply tests/drivers/build_all/bbram/testcase.yaml:10]: Fixed it.

File: tests/drivers/build_all/flash/testcase.yaml (hunk 4 above)
Comment (soburi) [reply tests/drivers/build_all/flash/testcase.yaml]: Fixed it.

File: tests/drivers/build_all/flash/src/main.c (hunk 3 above)
Comment (soburi) [reply tests/drivers/build_all/flash/src/main.c]: Fixed it.

File: tests/drivers/build_all/bbram/src/main.c (hunk 1 above)
Comment (soburi) [reply tests/drivers/build_all/bbram/src/main.c]: Thank you for your review.

I feel it is a bit weird that nothing a main() function, but I didn't have any problems with it, so I applied it.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Looks good, thanks. Please add the new `bbram` and `flash` tests to `MAINTAINERS.yml` to ensure we have the right review assignees and approvers for future PRs.

**Comment by soburi** [comment]
> Looks good, thanks. Please add the new `bbram` and `flash` tests to `MAINTAINERS.yml` to ensure we have the right review assignees and approvers for future PRs.

Thank you for the notification.
I updated the file.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**